saleor/core/utils/random_data.py

```diff
@@ -1639,3 +1643,23 @@ def get_product_list_images_dir(placeholder_dir):
 def get_image(image_dir, image_name):
     img_path = os.path.join(image_dir, image_name)
     return File(open(img_path, "rb"), name=image_name)
+
+
+def create_checkout_with_preorders():
+    channel = Channel.objects.get(currency_code="USD")
+    checkout = Checkout.objects.create(currency=channel.currency_code, channel=channel)
+    checkout.set_country(channel.default_country, commit=True)
+    checkout_info = fetch_checkout_info(checkout, [], [], get_plugins_manager())
+    for product_variant in ProductVariant.objects.all()[:2]:
+        product_variant.is_preorder = True
+        product_variant.preorder_global_threshold = 10
+        product_variant.preorder_end_date = timezone.now() + datetime.timedelta(days=10)
+        product_variant.save(
+            update_fields=[
+                "is_preorder",
+                "preorder_global_threshold",
+                "preorder_end_date",
+            ]
+        )
+        add_variant_to_checkout(checkout_info, product_variant, 2)
+    yield "Created checkout with two preorders"
```

Comment:
Maybe info with checkout token would be useful here? Assuming that you would like to reuse it.